src/routes/GroupExplorer/GroupSearch/GroupSearch.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react'
+import cx from 'classnames'
+import { useSelector } from 'react-redux'
+import Dropdown from 'components/Dropdown'
+import Icon from 'components/Icon'
+import Loading from 'components/Loading'
+import NoPosts from 'components/NoPosts'
+import ScrollListener from 'components/ScrollListener'
+import GroupCard from 'components/GroupCard'
+import useRouter from 'hooks/useRouter'
+import useDebounce from 'hooks/useDebounce'
+import useEnsureSearchedGroups from 'hooks/useEnsureSearchedGroups'
+import getMe from 'store/selectors/getMe'
+import { SORT_NAME, SORT_NEAREST, SORT_SIZE } from 'store/constants'
+import { CENTER_COLUMN_ID } from 'util/scrolling'
+import './GroupSearch.scss'
+
+export default function GroupSearch () {
+  const currentUser = useSelector(state => getMe(state))
+  const nearCoord = currentUser.locationObject ? { lng: parseFloat(currentUser.locationObject.center.lng), lat: parseFloat(currentUser.locationObject.center.lat) } : null
+  const membershipGroupIds = currentUser.memberships.toModelArray().map(membership => membership.group.id)
+  const [sortBy, setSortBy] = useState(SORT_NAME)
+  const [search, setSearch] = useState('')
+  const [offset, setOffset] = useState(0)
+  const debouncedSearchTerm = useDebounce(search, 500)
+  const { query } = useRouter()
+  const selectedGroupSlug = query.groupSlug
+  const { groups = [], pending = false, fetchMoreGroups, hasMore } = useEnsureSearchedGroups({ sortBy, search: debouncedSearchTerm, offset, nearCoord, visibility: [3] })
+  useEffect(() => {
```

Comment:
Style quibbles. I prefer a line break between the declarations and `useEffect` (et al) then a single line break between each of those. I'm pretty consistent about applying that pattern but we certainly don't have consensus on these things and matters little, so do as you will.